packages/xml-views-validation/test/test-utils.ts

```diff
@@ -30,3 +31,28 @@ export function testValidationsScenario(opts: {
   });
   opts.assertion(issues);
 }
+
+export function computeExpectedRanges(markedXMLSnippet: string): OffsetRange[] {
+  const expectedRanges = [];
+  const markedPattern = /🢂(.+?)🢀/gu;
+  let match;
+  // eslint-disable-next-line no-cond-assign
+  while ((match = markedPattern.exec(markedXMLSnippet))) {
+    const start = match.index;
+    // Chevrotain ranges are inclusive (start) to exclusive (end)
+    const end = start + match[1].length - 1;
+    expectedRanges.push({ start, end });
```

Comment:
I think you have to change the string you match against after the match. It won't work if there is more than 1 range in the string because of the offset caused by the arrows (which aren't in the arrow-less xml).

Reply:
Good point

Reply:
fixed